.github/codeql/queries/assert-pure.ql

```diff
@@ -1,21 +1,44 @@
 /**
  * @name Unwanted dependency on vscode API
- * @kind problem
+ * @kind path-problem
  * @problem.severity error
  * @id vscode-codeql/assert-pure
  * @description The modules stored under `pure` and tested in the `pure-tests`
  * are intended to be "pure".
  */
+
 import javascript
 
-class VSCodeImport extends ASTNode {
-  VSCodeImport() {
-    this.(Import).getImportedPath().getValue() = "vscode"
+class VSCodeImport extends AstNode {
+  VSCodeImport() { this.(Import).getImportedPath().getValue() = "vscode" }
+}
```

Comment:
Since you know the type you need, might as well declare it rather than casting.
```suggestion
class VSCodeImport extends Import {
  VSCodeImport() { this.getImportedPath().getValue() = "vscode" }
}
```

Reply:
Changing it to `Import` gets some errors because `Import` is an abstract class.
![Screenshot 2023-05-30 at 11 35 57](https://github.com/github/vscode-codeql/assets/3749000/c2cafe1e-6175-4ef2-bd24-642759dd4901)

I assume this was the reason it was like this originally. However, I should be able to change it to `ImportDeclaration` because all of the imports in our codebase are `import` rather than `require`. My understanding is that means we can use `ImportDeclaration` and not miss anything. At least right now it doesn't change the results.

Reply:
Oof I didn't realise `Import` was abstract.
Changing it to `extends ImportDeclaration` is a good specific solution.
If you want something more generic, try the (relatively new) `instanceof` keyword: `class VSCodeImport extends AstNode instanceof Import`. `instanceof` says we are creating a subtype of `Import` that inherits all its methods, but does not add to its set of values as an abstract class.